akka-actor/src/main/scala/akka/pattern/ReplyWithStatus.scala

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.pattern
+
+import scala.util.Try
+import scala.util.control.NoStackTrace
+import scala.util.{ Failure => ScalaFailure }
+import scala.util.{ Success => ScalaSuccess }
+
+/**
+ * Generic top level message type for signalling responses that may fail or succeed. Convenient to use together with the
+ * `askWithStatus` ask variants.
+ *
+ * Create using the factory methods [[ReplyWithStatus#succes]] and [[ReplyWithStatus#error]].
+ *
+ * Akka contains predefined serializers for the wrapper type and the textual error messages.
+ *
+ * @tparam T the type of value a successful reply would have
+ */
+final case class ReplyWithStatus[+T] private (private val status: Try[T]) {
```

Comment:
maybe not `case class`

Reply:
Yeah, I went back and forth on that and couldn't quite decide. Alternative is to implement toString, hashcode and equals since we want to be able to compare in tests and get something useful when logging.